src/Chapter2.hs

```diff
@@ -841,7 +851,10 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n xs
+  | n < 0 = []
+  | otherwise = take (length xs) $ drop n $ cycle xs
```

Comment:
Great!
If you want to speed up the solution, you can take `mod n (length xs)` 😉